ssl/t1_enc.c

```diff
@@ -375,6 +379,38 @@ int tls1_change_cipher_state(SSL *s, int which)
     if (ssl_get_max_send_fragment(s) != SSL3_RT_MAX_PLAIN_LENGTH)
         goto skip_ktls;
 
+# ifdef __FreeBSD__
+    memset(&crypto_info, 0, sizeof(crypto_info));
+    if (EVP_CIPHER_mode(c) == EVP_CIPH_GCM_MODE) {
+	 crypto_info.cipher_algorithm = CRYPTO_AES_NIST_GCM_16;
```

Comment:
How do we know we're using AES here?

Reply:
Hmm, is switching on the nid the best way to determine this?

Reply:
I tried switching on the NID and came up with a patch that worked, but duplicated some cases due to the "stitched" cipher suites.  As an alternative I have a version that does a switch on the algorithm_enc and algorithm_mac fields in the new cipher that is a bit more compact and easier to read.  I can go with whichever version you prefer.  You can compare the two by reviewing the diff that switches from the NID-based approach to the algorithm constants:  https://github.com/bsdjhb/openssl/commit/f507c3d2c0fb22494d2db65796717564b18e7b78

I did the changes on a side branch only to avoid adding noise to the PR.  Whichever solution you prefer I will merge into this commit.  I think it addresses your other questions in this file as well.

Reply:
I think switching on `algorithm_enc` and `algorithm_mac` looks like the way to go.